kusto/trustedEndpoints/trustedEndpoints.go

```diff
@@ -0,0 +1,240 @@
+		}
+		matchers[key] = f
+	}
+
+	return &TrustedEndpoints{matchers: matchers}
+}
+
+func (trusted *TrustedEndpoints) SetOverridePolicy(matcher func(string) bool) {
+	trusted.overrideMatcher = matcher
+}
+
+type TrustedEndpoints struct {
+	matchers          map[string]*FastSuffixMatcher
+	additionalMatcher *FastSuffixMatcher
+	overrideMatcher   func(string) bool
+}
+
+type MatchRule struct {
+	suffix string
+	exact  bool
+}
+
+type FastSuffixMatcher struct {
+	suffixLength int
+	rules        map[string][]MatchRule
+}
+
+func tailLowerCase(str string, length int) string {
+	if length <= 0 {
+		return ""
+	}
+
+	if length >= len(str) {
+		return strings.ToLower(str)
+	}
+
+	return strings.ToLower(str[len(str)-length:])
+}
+
+func (matcher *FastSuffixMatcher) isMatch(candidate string) bool {
+	if len(candidate) < matcher.suffixLength {
+		return false
+	}
+	if lst, ok := matcher.rules[tailLowerCase(candidate, matcher.suffixLength)]; ok {
+		for _, rule := range lst {
+			if strings.HasSuffix(strings.ToLower(candidate), rule.suffix) {
+				if len(candidate) == len(rule.suffix) || !rule.exact {
+					return true
+				}
+			}
+		}
+	}
+
+	return false
+}
+
+func NewFastSuffixMatcher(rules []MatchRule) (*FastSuffixMatcher, error) {
+	minSufLen := len(lo.MinBy(rules, func(a MatchRule, cur MatchRule) bool {
+		return len(a.suffix) < len(cur.suffix)
+	}).suffix)
+
+	if minSufLen == 0 || minSufLen == math.MaxInt32 {
+		return nil, errors.ES(
+			errors.OpUnknown,
+			errors.KClientArgs,
+			"FastSuffixMatcher should have at list one rule with at least one character",
+		).SetNoRetry()
+
+	}
+
+	processedRules := map[string][]MatchRule{}
+	for _, rule := range rules {
+		suffix := tailLowerCase(rule.suffix, minSufLen)
+		if lst, ok := processedRules[suffix]; !ok {
+			processedRules[suffix] = []MatchRule{rule}
+		} else {
+			processedRules[suffix] = append(lst, rule)
+		}
+	}
+
+	return &FastSuffixMatcher{
+		suffixLength: minSufLen,
+		rules:        processedRules,
+	}, nil
+}
+
+func values[T comparable, R any](m map[T]R) []R {
+	l := make([]R, 0, len(m))
+	for _, val := range m {
+		l = append(l, val)
+	}
+
+	return l
+}
+
+func CreateFastSuffixMatcherFromExisting(rules []MatchRule, existing *FastSuffixMatcher) (*FastSuffixMatcher, error) {
+	if existing == nil || len(existing.rules) == 0 {
+		return NewFastSuffixMatcher(rules)
+	}
+
+	if rules == nil || len(rules) == 0 {
+		return existing, nil
+	}
+
+	list := rules[:]
+
+	for _, elem := range values(existing.rules) {
+		list = append(list, elem...)
+	}
+
+	return NewFastSuffixMatcher(list)
+}
+
+func (trusted *TrustedEndpoints) AddTrustedHosts(rules []MatchRule, replace bool) error {
+	if rules == nil || len(rules) == 0 {
+		if replace {
+			trusted.additionalMatcher = nil
+		}
+		return nil
+	}
+
+	if replace {
+		trusted.additionalMatcher = nil
+	}
+
+	matcher, err := CreateFastSuffixMatcherFromExisting(rules, trusted.additionalMatcher)
+	trusted.additionalMatcher = matcher
+	return err
+}
+
+/**
```

Comment:
This is not how go doc strings work

Reply:
Close, but you need to add the name of the method to the start of the doc: 
`// AddTrustedHosts Add or set a list of trusted endpoints rules`
`func (trusted *TrustedEndpoints) AddTrustedHosts(rules []MatchRule, replace bool) error {`

I'm surprised whatever plugin you use doesn't  auto correct this

Reply:
yea could be nice